lib/sles4sap_publiccloud.pm

```diff
@@ -667,4 +673,96 @@ sub create_hana_vars_section {
     return (\%hana_vars);
 }
 
+=head2 get_hana_database_info
+
+    Parses "hdbsql" command output.
+    Returns 1 if hana database is online otherwise 0 offline.
+
+=back
+=cut
+
+sub get_hana_database_info {
+    my ($self, $password_db, $instance_id) = @_;
+    my $hana_database_online = '"HDB","HDB-$instance_id","YES","ACTIVE","","","DEFAULT","?"';
+
+    sleep 30;
+    eval {
+        my $output_cmd = $self->run_cmd(cmd => "hdbsql -u SYSTEM -p $password_db -i $instance_id 'SELECT * FROM SYS.M_DATABASES;'", runas => get_required_var("SAP_SIDADM"));
+    };
+    if ($@) {
+        record_info('HANA DB OFFLINE', "Hana database in primary node is offline. Here the error \n$@");
+        return 0;
+    }
+    else {
+        return 1;
+    }
+}
+
+=head2 is_hana_database_offline
+
+    Check if hana DB is offline.
+    Returns if hana database is online with 1 and offline with 0
+
+=over 1
+
+=item B<TIMEOUT> - default 900
+
+=back
+=cut
+
+sub is_hana_database_offline {
+    my ($self, %args) = @_;
+    my $timeout = bmwqemu::scale_timeout($args{timeout} // 900);
+    my $start_time = time;
+    my $db_status = -1;
+    my $consecutive_passes = 0;
+    my $password_db = "Linux1234";
+    my $instance_id = get_required_var('INSTANCE_ID');
+    my $total_consecutive_passes = ($args{total_consecutive_passes} // 5);
+
+    while ($consecutive_passes < $total_consecutive_passes) {
+        $db_status = $self->get_hana_database_info($password_db, $instance_id);
+        if (time - $start_time > $timeout) {
+            record_info("Hana database after timeout", $self->run_cmd(cmd => "hdbsql -u SYSTEM -p $password_db -i $instance_id 'SELECT * FROM SYS.M_DATABASES;'"));
+            die("Hana database is still online");
+        }
+        if ($db_status == 0) {
+            last;
+        }
+        sleep 30;
+        ++$consecutive_passes;
+    }
+    return $db_status;
+}
+
+=head2 is_primary_node_offline
+
+    Check if primary node is offline.
+    Returns if primary node is offline with 0 and 1 online
+
+=back
+=cut
+
+sub is_primary_node_offline {
+    my ($self, %args) = @_;
+    my $sapadm = lc(get_required_var('INSTANCE_SID')) . 'adm';
+
+    # Wait by default for 5 minutes
+    my $time_to_wait = 300;
+    my $cmd = "su - $sapadm -c 'python exe/python_support/systemReplicationStatus.py'";
+
+    # Loop until is not primary the vm01 or timeout is reached
+    while ($time_to_wait > 0) {
+        my $output = script_output($cmd, proceed_on_failure => 1);
+        if ($output !~ /mode:[\r\n\s]+PRIMARY/) {
+            record_info('SYSTEM REPLICATION STATUS', "System replication status in pimary node.\n$output");
+            return 0;
+        }
+
+        $time_to_wait -= 10;
+        sleep 10;
+    }
+    return 1;
```

Comment:
Similar to above, but it might be for broader discussion maybe. 
I guess it depends what approach you take when reading the code. Either RC or true/false. 
But behavior is documented so if somebody gets confused they will find it.